### ISSUE TITLE
Error parsing C-style strings

### ISSUE THREAD

**Issue by benjie** [vitaly-t/pg-minify#2]
PostgreSQL supports C-style string escaping in `E''` strings:

https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-STRINGS-ESCAPE

e.g.:

```sql
# select E'\'' as example;
 example
---------
 '
(1 row)

Time: 0.241 ms
```

However pg-minify v1.0.0 (and v0.5.5) baulks at this:

```js
> m = require('pg-minify')
[Function: minify] {
  SQLParsingError: [Function: SQLParsingError],
  parsingErrorCode: {
    unclosedMLC: 0,
    unclosedText: 1,
    unclosedQI: 2,
    multiLineQI: 3,
    nestedMLC: 4
  }
}
> m("select E'\'' as example;")
Thrown:
SQLParsingError: 1
    at new SQLParsingError (/Users/benjiegillam/Dev/graphile-migrate/node_modules/pg-minify/lib/error.js:25:24)
    at throwError (/Users/benjiegillam/Dev/graphile-migrate/node_modules/pg-minify/lib/parser.js:177:15)
    at minify (/Users/benjiegillam/Dev/graphile-migrate/node_modules/pg-minify/lib/parser.js:113:17)
    at repl:1:1
    at Script.runInThisContext (vm.js:123:20)
    at REPLServer.defaultEval (repl.js:358:29)
    at bound (domain.js:415:14)
    at REPLServer.runBound [as eval] (domain.js:428:12)
    at REPLServer.onLine (repl.js:665:10)
    at REPLServer.emit (events.js:201:15) {
  name: 'SQLParsingError',
  stack: 'SQLParsingError: 1\n    at new SQLParsingError ' +
    '(/Users/benjiegillam/Dev/graphile-migrate/node_modules/pg-minify/lib/error.js:25:24)\n' +
    '    at throwError ' +
    '(/Users/benjiegillam/Dev/graphile-migrate/node_modules/pg-minify/lib/parser.js:177:15)\n' +
    '    at minify ' +
    '(/Users/benjiegillam/Dev/graphile-migrate/node_modules/pg-minify/lib/parser.js:113:17)\n' +
    '    at repl:1:1\n    at Script.runInThisContext (vm.js:123:20)\n    at ' +
    'REPLServer.defaultEval (repl.js:358:29)\n    at bound ' +
    '(domain.js:415:14)\n    at REPLServer.runBound [as eval] ' +
    '(domain.js:428:12)\n    at REPLServer.onLine (repl.js:665:10)\n    at ' +
    'REPLServer.emit (events.js:201:15)',
  code: 1,
  error: 'Unclosed text block.',
  position: { line: 1, column: 9 },
  message: 'Error parsing SQL at {line:1,col:9}: Unclosed text block.'
}
>
```

**Comment by vitaly-t** [comment]
Yes, this seems like a special case that's not supported at the moment.

The library expects that `\'` is escaped as `''`, which then works.

It will require some tweaking to make the reported case work also.

**Comment by vitaly-t** [comment]
I'm trying to come up with a RegEx that would help me skip those special cases, without complicating the algorithm further. I had to open a [question on StackOverflow for it](https://stackoverflow.com/questions/56046843/skip-odd-number-of-preceding-symbols-when-searching), as I got stuck with such a RegEx.

We just need to skip every `'` that's preceded by an odd number of `\`.

**Comment by vitaly-t** [comment]
Fixed in [v1.1.0](https://github.com/vitaly-t/pg-minify/releases/tag/1.1.0), and in much a simpler way, without any RegEx.

@benjie Thanks for reporting!

**Comment by vitaly-t** [comment]
@benjie Release [1.2.0](https://github.com/vitaly-t/pg-minify/releases/tag/1.2.0) now makes this library way better.
